Leo-Chen-CM/ARGO-23
# Feat/fuzzy logic

Finally made this work lessgo!

I created a fuzzytest script that gets passed a weight calculated in bothandler. The fuzzytest script takes the weight it got handed and checks it against three separate animation curves that represent the set data for movement.

 We call an evaluate function that's part of the animationcurve component to see which set the weight is most compatible with. This dictates our bots behaviour.

Over in bothandler we have an anchor weight set to .5 which gets added to/subtracted from by a differential variable(df). Depending on our bots chosen difficulty we have different probabilities that dt will be decremented from. If dt gets to low our bots will be pulled towards not moving and colliding with an obstacle which will kill the bot.

Steps to test:
- set the bots difficulty in our bothandler script to easy.
- run the game and allow it to run for awhile. The differential variable (accessible in  inspector) should decrement over time.
- once the differential hits our low threshold the bot should stop moving and accept it's fate.
- do the same for each difficulty and we should expect the same end point though over a greater time scale.

Notice anything weird or unclear and do let me know.
